client/state/data-layer/wpcom/sites/comment-counts/index.js

```diff
@@ -0,0 +1,41 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+// import { translate } from 'i18n-calypso';
+// import { flatMap, flatten, isArray, map } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { COMMENT_COUNTS_REQUEST } from 'state/action-types';
+import { mergeHandlers } from 'state/action-watchers/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+
+export const fetchCommentCounts = ( { dispatch }, action ) => {
+	const { siteId } = action.query;
+	const postId = action.query.postId || null;
+
+	dispatch(
+		http(
+			{
+				method: 'GET',
+				path: `/sites/${ siteId }/comment-counts`,
+				apiVersion: '1.0',
+				query: {
+					postId,
+				},
+			},
+			action
+		)
+	);
+};
+
+const treeHandlers = {
+	[ COMMENT_COUNTS_REQUEST ]: [ dispatchRequest( fetchCommentCounts ) ],
```

Comment:
We can use the new `dispatchRequestEx` which changes the signature of the handlers to return an action or action thunk.  (It'll eventually be renamed back to dispatchRequest once all instances are updated). 

See p4TIVU-8LU-p2 for a full example.